src/isar/robot/robot.py

```diff
@@ -0,0 +1,261 @@
+                        error_description = (
+                            f"Mission will be cancelled after failing to initiate "
+                            f"{settings.INITIATE_FAILURE_COUNTER_LIMIT} times because: "
+                            f"{e.error_description}"
+                        )
+
+                        self._trigger_event(
+                            self.queues.robot_mission_failed,
+                            ErrorMessage(
+                                error_reason=e.error_reason,
+                                error_description=error_description,
+                            ),
+                        )
+                started_mission = True
+        except RobotInfeasibleTaskException as e:
+            error_description = (
+                f"Failed to initiate task "
+                f"{str(mission_or_task.id)[:8]} after retrying "
+                f"{retries} times because: "
+                f"{e.error_description}"
+            )
+            self._trigger_event(
+                self.queues.robot_task_status_failed,
+                ErrorMessage(
+                    error_reason=e.error_reason, error_description=error_description
+                ),
+            )
+        except RobotInfeasibleMissionException as e:
+            self._trigger_event(
+                self.queues.robot_mission_failed,
+                ErrorMessage(
+                    error_reason=e.error_reason, error_description=e.error_description
+                ),
+            )
+        self.robot_task_status_thread = ThreadedRequest(
+            request_func=self.run_robot_task_status_thread
+        )
+        self.robot_task_status_thread.start_thread(name="Robot task status thread")
+
+    def handle_stop_mission(self):
+        return NotImplementedError()
+
+    def handle_pause_mission(self):
+        return NotImplementedError()
+
+    def run(self) -> None:
+        self.robot_status_thread = ThreadedRequest(
+            request_func=self.run_robot_status_thread
+        )
+        self.robot_status_thread.start_thread(name="Robot status thread")
+        while True:
+            start_mission_or_task = self._check_for_event(
+                self.queues.state_machine_start_mission
+            )
+            if start_mission_or_task != None:
+                if (
+                    self.start_mission_thread != None
+                    and self.start_mission_thread._is_thread_alive()
+                ):
+                    self.start_mission_thread.wait_for_thread()
+                self.start_mission_thread = ThreadedRequest(
+                    request_func=self.run_start_mission_thread,
+                )
+                self.start_mission_thread.start_thread(
+                    start_mission_or_task, name="Robot start mission thread"
+                )
+                # TODO: consider what happens when we get another start_mission while the start_mission_thread is running.
```

Comment:
Remove todo

Reply:
Will be removed before moving the PR out of draft